### PR TITLE
xi-unicode can be marked as no_std!

### PR DESCRIPTION
**Pull request by mooman219**
## Summary
This PR marks the library as `no_std` since all of its `std` API usage exists in `core`. This is helpful for embedding in environments where `std` isn't available, like the web. 

## Related Issues
Closes: https://github.com/xi-editor/xi-editor/issues/1250

## Review Checklist

- [ ] I have responded to reviews and made changes where appropriate.
- [x] I have tested the code with `cargo test --all` / `./rust/run_all_checks`.
- [x] I have updated comments / documentation related to the changes I made.
- [x] I have rebased my PR branch onto xi-editor/master.

### CODE REVIEW

**Comment by mooman219** [comment]
```
    Checking xi-core-lib v0.3.0 (/home/travis/build/xi-editor/xi-editor/rust/core-lib)
error: use of deprecated item 'std::error::Error::description': use the Display impl or to_string()
   --> core-lib/src/config.rs:697:62
    |
697 |             UnknownDomain(ref s) => write!(f, "{}: {}", self.description(), s),
    |                                                              ^^^^^^^^^^^
    |
    = note: `-D deprecated` implied by `-D warnings`
error: use of deprecated item 'std::error::Error::description': use the Display impl or to_string()
   --> core-lib/src/config.rs:698:70
    |
698 |             Parse(ref p, ref e) => write!(f, "{} ({:?}), {:?}", self.description(), p, e),
    |                                                                      ^^^^^^^^^^^
error: use of deprecated item 'std::error::Error::description': use the Display impl or to_string()
   --> core-lib/src/config.rs:710:34
    |
710 |             Parse(_, ref e) => e.description(),
    |                                  ^^^^^^^^^^^
error: use of deprecated item 'std::error::Error::description': use the Display impl or to_string()
   --> core-lib/src/config.rs:711:28
    |
711 |             Io(ref e) => e.description(),
    |                            ^^^^^^^^^^^
error: use of deprecated item 'std::error::Error::description': use the Display impl or to_string()
   --> core-lib/src/config.rs:712:40
    |
712 |             UnexpectedItem(ref e) => e.description(),
    |                                        ^^^^^^^^^^^
error: aborting due to 5 previous errors
error: could not compile `xi-core-lib`.
```

Failed for unrelated reasons.

**Comment by cmyr** [comment]
fixed the lints, feel free to rebase

**Comment by mooman219** [comment]
Done

File: rust/unicode/src/lib.rs [hunk 1]
```diff
@@ -13,9 +13,13 @@
...
 
 //! Unicode utilities useful for text editing, including a line breaking iterator.
+#![no_std]
+
+extern crate alloc;
+
 mod tables;
 
-use std::cmp::Ordering;
+use core::cmp::Ordering;
```
Comment (cmyr) [inline rust/unicode/src/lib.rs]: I haven't done any `no_std` stuff before; is it standard to always use core, or do you use `std` when it's available? Do we want to have a feature? That's [what serde does](https://github.com/serde-rs/serde/blob/master/serde/src/lib.rs#L80) but I'm not sure about the rationale?

Comment (jansol) [reply rust/unicode/src/lib.rs]: AFAIK you just use core, since std just re-exports that anyway.

Comment (cmyr) [reply rust/unicode/src/lib.rs]: sgtm!

Comment (mooman219) [reply rust/unicode/src/lib.rs]: +1 jansol, `std` over simplified is just re-exported core + less portable additional features bundled together.

**Pull request merged**